esphome/components/alarm_control_panel/__init__.py

```diff
@@ -189,9 +189,11 @@ async def setup_alarm_control_panel_core_(var, config):
     for conf in config.get(CONF_ON_READY, []):
         trigger = cg.new_Pvariable(conf[CONF_TRIGGER_ID], var)
         await automation.build_automation(trigger, [], conf)
-    if (webserver_id := config.get(CONF_WEB_SERVER_ID)) is not None:
+    if (web_server_config := config.get("web_server")) is not None and (
+        webserver_id := web_server_config.get(CONF_WEB_SERVER_ID)
+    ) is not None:
         web_server_ = await cg.get_variable(webserver_id)
-        web_server.add_entity_to_sorting_list(web_server_, var, config)
+        web_server.add_entity_to_sorting_list(web_server_, var, web_server_config)
```

Comment:
Lets rename this function to be more generic, then if we need to add things later these files won't need changing again.
```suggestion
        web_server.add_entity_config(web_server_, var, web_server_config)
```